smithy-aws-traits/src/main/java/software/amazon/smithy/aws/traits/tagging/TaggableTrait.java

```diff
@@ -0,0 +1,173 @@
+     *
+     * @return Return the ShapeId of the operation specified for the Tag API.
+     */
+    public Optional<ShapeId> getTagApi() {
+        return Optional.ofNullable(tagApi);
+    }
+
+    /**
+     * Gets the rame of the property that represents the tags on the resource, if any.
+     *
+     * @return Return the name of the property that represents the tags of the resource.
+     */
+    public Optional<String> getProperty() {
+        return Optional.ofNullable(property);
+    }
+
+    @Override
+    protected Node createNode() {
+        return new ObjectNode(MapUtils.of(), getSourceLocation())
+                .withOptionalMember("property", getProperty().map(Node::from))
+                .withOptionalMember("tagApi", getTagApi().map(id -> Node.from(id.toString())))
+                .withOptionalMember("untagApi", getUntagApi().map(id -> Node.from(id.toString())))
+                .withOptionalMember("listTagsApi", getListTagsApi().map(id -> Node.from(id.toString())))
+                .withOptionalMember("supportsSystemTags", getSupportsSystemTags().map(Node::from));
+    }
+
+    public static Builder builder() {
+        return new Builder();
+    }
+
+    @Override
+    public SmithyBuilder<TaggableTrait> toBuilder() {
+        return builder()
+                .property(property)
+                .tagApi(tagApi)
+                .untagApi(untagApi)
+                .listTagsApi(listTagsApi)
+                .supportsSystemTags(supportsSystemTags)
+                .sourceLocation(getSourceLocation());
+    }
+
+    public static final class Builder extends AbstractTraitBuilder<TaggableTrait, Builder> {
+        private String property;
+        private ShapeId tagApi;
+        private ShapeId untagApi;
+        private ShapeId listTagsApi;
+        private Boolean supportsSystemTags;
+
+        public Builder property(String property) {
+            this.property = property;
+            return this;
+        }
+
+        public Builder tagApi(ShapeId tagApi) {
+            this.tagApi = tagApi;
+            return this;
+        }
+
+        public Builder untagApi(ShapeId untagApi) {
+            this.untagApi = untagApi;
+            return this;
+        }
+
+        public Builder listTagsApi(ShapeId listTagsApi) {
+            this.listTagsApi = listTagsApi;
+            return this;
+        }
+
+        public Builder supportsSystemTags(Boolean supportsSystemTags) {
+            this.supportsSystemTags = supportsSystemTags;
+            return this;
+        }
+
+        @Override
+        public TaggableTrait build() {
+            return new TaggableTrait(this);
+        }
+    }
+
+    public static final class Provider implements TraitService {
+        @Override
+        public ShapeId getShapeId() {
+            return ID;
+        }
+
+        @Override
+        public TaggableTrait createTrait(ShapeId target, Node value) {
+            NodeMapper nodeMapper = new NodeMapper();
```

Comment:
Is NodeMapper needed here? Seems simple enough to implement.

Reply:
I was guided to use NodeMapper in a prior comment (other PR) by `@kstich.`

https://github.com/awslabs/smithy/pull/1272#discussion_r897368893
> Can this use a `NodeMapper` instead?

I understand the merits of doing it either way for performance or code maintenance, but I'm getting an inconsistent nudges.